other/restart-deployment-on-secret-change/06-check-restart.yaml

```diff
@@ -0,0 +1,4 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestStep
+commands:
+- script: "if [ \"$(kubectl get pods -n default | grep busybox | awk 'NR==1{print $1}')\" != \"$(cat pod-name.txt)\" ];then exit;else exit 1;fi"
```

Comment:
You could have also compared the resourceVersion in the modified secret with the value of the `ops.corp.com/triggerrestart` but this is also ok :)

Reply:
Sounds like a better idea. I'll try this out.

Reply:
So I'm not sure if the annotation is getting removed after the restart, I'm not able to get the annotation `ops.corp.com/trigger`. It's just isn't there. But I can see the annotation `deployment.kubernetes.io/revision` getting incremented everytime a restart happens. Should I check with that?

Reply:
I still had to save the `resourceVersion` of the old secret in a text file and then take it to compare with the annotation. But it was fun! Realized that the `resourceVersion` gets updated only after the kyverno admission check.